access-widener-javaparser/src/main/java/net/fabricmc/accesswidener/javaparser/SourceUtil.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2020 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.accesswidener.javaparser;
+
+import java.util.Iterator;
+import java.util.List;
+
+import com.github.javaparser.ast.Modifier;
+import com.github.javaparser.ast.body.TypeDeclaration;
+import org.objectweb.asm.Opcodes;
+
+class SourceUtil implements Opcodes {
+	static void fromFlags(List<Modifier> modifiers, int access, boolean abstract_) {
+		for (int i = 0; i < 32; i++) {
+			int flag = 1 << i;
+
+			switch (flag & access) {
+			case ACC_ABSTRACT:
+				if (!abstract_) {
+					modifiers.add(Modifier.abstractModifier());
+				}
```

Comment:
shouldn't an else branch add the default modifier here?

Reply:
well no, because it wouldn't be triggered if the method was abstract, I've renamed the local and fixed this case in a different way

Reply:
but if you have an interface method without acc_abstract, it needs either default or static?